### ISSUE TITLE
OpenWrt/Mikrotik  v0.4.5  panic: runtime error

### ISSUE THREAD

**Issue by DrewCyber** [yggdrasil-network/yggdrasil-go#962]
Openwrt binary was built on mac with command below and uploaded to mikrotik. From time to time yggdrasil process fails with panic error.
Previous (0.4.4) version before was built with same command and worked fine.

Build string:
`
GOOS=linux GOARCH=mips GOMIPS=softfloat go build $ARGS -ldflags="$LDFLAGS" -gcflags="$GCFLAGS" ./cmd/$CMD
`

```
root@OpenWrt:~# yggdrasilctl  getself
Build name:   	yggdrasil-builder4mibs
Build version:	0.4.5-1-g373280e
IPv6 address: 	200:fb97:f9b6:38cb:9c3:f854:98ec:72fa
IPv6 subnet:  	300:fb97:f9b6:38cb::/64
Coordinates:  	[1 1 11 310 11 28]
Public key:   	82340324e39a7b1e03d5b389c682e5d803f7884ac4dab8e7854b77260dd3dab8
```
Openwrt details:
```
"kernel": "4.14.241",
"hostname":"OpenWrt",
"system": "Qualcomm Atheros QCA9533 ver 2 rev 0",
"model": "MikroTik RouterBOARD 952Ui-5ac2nD",
"board_name": "rb-952ui-5ac2nd"}'
```

Fail log 1:

```
Thu Oct 20 15:00:14 2022 daemon.err yggdrasil[2274]: panic: runtime error: invalid memory address or nil pointer dereference
Thu Oct 20 15:00:14 2022 daemon.err yggdrasil[2274]: [signal SIGSEGV: segmentation violation code=0x1 addr=0xc pc=0x2ac540]
Thu Oct 20 15:00:14 2022 daemon.err yggdrasil[2274]:
Thu Oct 20 15:00:14 2022 daemon.err yggdrasil[2274]: goroutine 61485 [running]:
Thu Oct 20 15:00:14 2022 daemon.err yggdrasil[2274]: github.com/yggdrasil-network/yggdrasil-go/src/core.(*Core).GetPeers.func1()
Thu Oct 20 15:00:14 2022 daemon.err yggdrasil[2274]: 	/Users/cyber/Documents/github/yggdrasil-go/src/core/api.go:65 +0xa4
Thu Oct 20 15:00:14 2022 daemon.err yggdrasil[2274]: github.com/Arceliar/phony.Block.func1()
Thu Oct 20 15:00:14 2022 daemon.err yggdrasil[2274]: 	/Users/cyber/go/pkg/mod/github.com/!arceliar/phony@v0.0.0-20210209235338-dde1a8dca979/actor.go:80 +0x44
Thu Oct 20 15:00:14 2022 daemon.err yggdrasil[2274]: github.com/Arceliar/phony.(*Inbox).run(0x87a334)
Thu Oct 20 15:00:14 2022 daemon.err yggdrasil[2274]: 	/Users/cyber/go/pkg/mod/github.com/!arceliar/phony@v0.0.0-20210209235338-dde1a8dca979/actor.go:89 +0x68
Thu Oct 20 15:00:14 2022 daemon.err yggdrasil[2274]: created by github.com/Arceliar/phony.(*Inbox).restart
Thu Oct 20 15:00:14 2022 daemon.notice netifd: Network device 'ygg0' link is down
Thu Oct 20 15:00:14 2022 daemon.notice netifd: Interface 'yggdrasil' has link connectivity loss
Thu Oct 20 15:00:14 2022 daemon.notice netifd: Interface 'yggdrasil' is now down
Thu Oct 20 15:00:14 2022 daemon.err yggdrasil[2274]: 	/Users/cyber/go/pkg/mod/github.com/!arceliar/phony@v0.0.0-20210209235338-dde1a8dca979/actor.go:121 +0x50
Thu Oct 20 15:00:16 2022 daemon.notice netifd: Interface 'yggdrasil' is disabled
```
Fail log 2:
```
Thu Oct 20 21:54:36 2022 daemon.err yggdrasil[5126]: panic: runtime error: invalid memory address or nil pointer dereference
Thu Oct 20 21:54:36 2022 daemon.err yggdrasil[5126]: [signal SIGSEGV: segmentation violation code=0x1 addr=0xc pc=0x2ac540]
Thu Oct 20 21:54:36 2022 daemon.err yggdrasil[5126]:
Thu Oct 20 21:54:36 2022 daemon.err yggdrasil[5126]: goroutine 88336 [running]:
Thu Oct 20 21:54:36 2022 daemon.err yggdrasil[5126]: github.com/yggdrasil-network/yggdrasil-go/src/core.(*Core).GetPeers.func1()
Thu Oct 20 21:54:36 2022 daemon.err yggdrasil[5126]: 	/Users/cyber/Documents/github/yggdrasil-go/src/core/api.go:65 +0xa4
Thu Oct 20 21:54:36 2022 daemon.err yggdrasil[5126]: github.com/Arceliar/phony.Block.func1()
Thu Oct 20 21:54:36 2022 daemon.err yggdrasil[5126]: 	/Users/cyber/go/pkg/mod/github.com/!arceliar/phony@v0.0.0-20210209235338-dde1a8dca979/actor.go:80 +0x44
Thu Oct 20 21:54:36 2022 daemon.err yggdrasil[5126]: github.com/Arceliar/phony.(*Inbox).run(0xc82334)
Thu Oct 20 21:54:36 2022 daemon.err yggdrasil[5126]: 	/Users/cyber/go/pkg/mod/github.com/!arceliar/phony@v0.0.0-20210209235338-dde1a8dca979/actor.go:89 +0x68
Thu Oct 20 21:54:36 2022 daemon.err yggdrasil[5126]: created by github.com/Arceliar/phony.(*Inbox).restart
Thu Oct 20 21:54:36 2022 daemon.err yggdrasil[5126]: 	/Users/cyber/go/pkg/mod/github.com/!arceliar/phony@v0.0.0-20210209235338-dde1a8dca979/actor.go:121 +0x50
Thu Oct 20 21:54:36 2022 daemon.notice netifd: Network device 'ygg0' link is down
```

**Comment by DrewCyber** [comment]
Looks like fixed in 0.4.6

**Comment by neilalexander** [comment]
Closing as fixed.
